bundles/org.openhab.binding.homewizard/src/main/java/org/openhab/binding/homewizard/internal/HomeWizardHandler.java

```diff
@@ -0,0 +1,274 @@
+     */
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        Object cfgObject = configurationParameters.get("refreshDelay");
+        if (cfgObject != null) {
+            BigDecimal bd = (BigDecimal) cfgObject;
+            config.refreshDelay = bd.intValue();
+        }
+
+        cfgObject = configurationParameters.get("ipAddress");
+        if (cfgObject != null) {
+            config.ipAddress = (String) cfgObject;
+        }
+
+        if (configure()) {
+            // If the new configuration is proper, stop the poller if needed
+            stopPolling(true);
+            // Then start it again if it was stopped
+            startPolling();
+        } else {
+            // Stop polling if the new config is invalid
+            stopPolling(false);
+        }
+    }
+
+    /**
+     * Check the current configuration
+     *
+     * @return true if the configuration is ok to start polling, false otherwise
+     */
+    private boolean configure() {
+        if (config.ipAddress.trim().isEmpty()) {
+            // since it is marked as required, initialize() should not be called if this field is empty
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing ipAddress/host configuration");
+            return false;
+        } else {
+            updateStatus(ThingStatus.UNKNOWN);
+
+            try {
+                configurationLock.lock();
+                apiURL = String.format("http://%s/api/v1/data", config.ipAddress.trim());
+            } finally {
+                configurationLock.unlock();
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Stop the poller unconditionally
+     */
+    @Override
+    public void dispose() {
+        stopPolling(false);
+    }
+
+    /**
+     * Stop the polling job
+     *
+     * @param onlyIfNeeded if true polling will only actually stop if a new refresh interval should be set
+     */
+    private void stopPolling(boolean onlyIfNeeded) {
+        try {
+            pollingJobLock.lock();
+            if (pollingJob != null && !pollingJob.isCancelled()) {
+                if (!onlyIfNeeded || config.refreshDelay != currentRefreshDelay) {
+                    if (pollingJob != null) {
+                        pollingJob.cancel(true);
+                    }
+                    pollingJob = null;
+                }
+            }
+        } finally {
+            pollingJobLock.unlock();
+        }
+    }
+
+    /**
+     * Start a polling job if it is not already running.
+     */
+    private void startPolling() {
+        try {
+            pollingJobLock.lock();
+            boolean startPoller = false;
+            if (pollingJob != null) {
+                startPoller = pollingJob.isCancelled();
+            } else {
+                startPoller = true;
+            }
+
+            if (startPoller) {

```

Comment:
This is a bit over complicated. If start is called and you only do this from initialize we can assume no poller is running. So you can just start here without any checks or locks.